src/Features/CSharp/Portable/ConvertForEachToFor/CSharpConvertForEachToForCodeRefactoringProvider.cs

```diff
@@ -25,24 +26,9 @@ public CSharpConvertForEachToForCodeRefactoringProvider()
 
         protected override string Title => CSharpFeaturesResources.Convert_to_for;
 
-        protected override ForEachStatementSyntax GetForEachStatement(TextSpan selection, SyntaxToken token)
-        {
-            var foreachStatement = token.Parent.FirstAncestorOrSelf<ForEachStatementSyntax>();
-            // https://github.com/dotnet/roslyn/issues/30584: Add tests for this scenario
-            if (foreachStatement == null || foreachStatement.AwaitKeyword != default)
-            {
-                return null;
-            }
-
-            // support refactoring only if caret is in between "foreach" and ")"
-            var scope = TextSpan.FromBounds(foreachStatement.ForEachKeyword.Span.Start, foreachStatement.CloseParenToken.Span.End);
-            if (!scope.IntersectsWith(selection))
-            {
-                return null;
-            }
-
-            return foreachStatement;
-        }
+        // https://github.com/dotnet/roslyn/issues/30584: Add tests for this scenario
```

Comment:
have you done this? :)  (i know it's not your comment).

Reply:
I'll look into it. 

> See comment in CSharpConvertForEachToForCodeRefactoringProvider.GetForEachStatement.

Also, is this some kind of puzzle where I have to follow links across codebase and GitHub?